File: src/test/java/com/github/ggerla/java2csharp/testclasses/admin/Information.java
/*
 * Copyright 2014 Giuseppe Gerla. All Rights Reserved.
 *
 * Licensed under the GNU General Public License, Version 2.0;
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.gnu.org/licenses/gpl2.txt
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.github.ggerla.java2csharp.testclasses.admin;

import java.util.List;

public class Information {
	private int id;

	private int defaultRole;

	private String name;

	private String surname;

	private String idNumber;

	private String telephone;

	private String mobile;

	private String fax;

	private String email;

	private String department;

	private String address;

	private boolean enabled;

	private String username;

	private String password;

	private List<Integer> roles;

	public Information(int id, int defaultRole,
			String name, String surname, String idNumber, String telephone,
			String mobile, String fax, String email, String department,
			String address, boolean enabled, String username, String password,
			List<Integer> roles) {
		super();
		this.id = id;
		this.defaultRole = defaultRole;
		this.name = name;
		this.surname = surname;
		this.idNumber = idNumber;
		this.telephone = telephone;
		this.mobile = mobile;
		this.fax = fax;
		this.email = email;
		this.department = department;
		this.address = address;
		this.enabled = enabled;
		this.username = username;
		this.password = password;
		this.roles = roles;
	}

	protected Information() {
		super();
	}

	public int getId() {
		return id;
	}

	public void setId(int id) {
		this.id = id;
	}

	public int getDefaultRole() {
		return defaultRole;
	}

	public void setDefaultRole(int defaultRole) {
		this.defaultRole = defaultRole;
	}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

	public String getSurname() {
		return surname;
	}

	public void setSurname(String surname) {
		this.surname = surname;
	}

	public String getIdNumber() {
		return idNumber;
	}

	public void setIdNumber(String idNumber) {
		this.idNumber = idNumber;
	}

	public String getTelephone() {
		return telephone;
	}

	public void setTelephone(String telephone) {
		this.telephone = telephone;
	}

	public String getMobile() {
		return mobile;
	}

	public void setMobile(String mobile) {
		this.mobile = mobile;
	}

	public String getFax() {
		return fax;
	}

	public void setFax(String fax) {
		this.fax = fax;
	}

	public String getEmail() {
		return email;
	}

	public void setEmail(String email) {
		this.email = email;
	}

	public String getDepartment() {
		return department;
	}

	public void setDepartment(String department) {
		this.department = department;
	}

	public String getAddress() {
		return address;
	}

	public void setAddress(String address) {
		this.address = address;
	}

	public boolean isEnabled() {
		return enabled;
	}

	public void setEnabled(boolean enabled) {
		this.enabled = enabled;
	}

	public String getUsername() {
		return username;
	}

	public void setUsername(String username) {
		this.username = username;
	}

	public String getPassword() {
		return password;
	}

	public void setPassword(String password) {
		this.password = password;
	}

	public List<Integer> getRoles() {
		return roles;
	}

	public void setRoles(List<Integer> roles) {
		this.roles = roles;
	}
}

File: src/test/resources/expectedfolder/admin/Information.cs
using System;
using System.Collections.Generic;

namespace org.codejuicer.java2csharp.testclasses.admin
{
	public class Information
	{
		private int id;

		private int defaultRole;

		private string name;

		private string surname;

		private string idNumber;

		private string telephone;

		private string mobile;

		private string fax;

		private string email;

		private string department;

		private string address;

		private bool enabled;

		private string username;

		private string password;

		private IList<int?> roles;

		public Information(int id, int defaultRole, string name, string surname, string idNumber
			, string telephone, string mobile, string fax, string email, string department, 
			string address, bool enabled, string username, string password, IList<int?> roles
			)
			: base()
		{
			this.id = id;
			this.defaultRole = defaultRole;
			this.name = name;
			this.surname = surname;
			this.idNumber = idNumber;
			this.telephone = telephone;
			this.mobile = mobile;
			this.fax = fax;
			this.email = email;
			this.department = department;
			this.address = address;
			this.enabled = enabled;
			this.username = username;
			this.password = password;
			this.roles = roles;
		}

		protected internal Information()
			: base()
		{
		}

		public virtual int Id
		{
			get
			{
				return id;
			}
			set
			{
				int id = value;
				this.id = id;
			}
		}

		public virtual int DefaultRole
		{
			get
			{
				return defaultRole;
			}
			set
			{
				int defaultRole = value;
				this.defaultRole = defaultRole;
			}
		}

		public virtual string Name
		{
			get
			{
				return name;
			}
			set
			{
				string name = value;
				this.name = name;
			}
		}

		public virtual string Surname
		{
			get
			{
				return surname;
			}
			set
			{
				string surname = value;
				this.surname = surname;
			}
		}

		public virtual string IdNumber
		{
			get
			{
				return idNumber;
			}
			set
			{
				string idNumber = value;
				this.idNumber = idNumber;
			}
		}

		public virtual string Telephone
		{
			get
			{
				return telephone;
			}
			set
			{
				string telephone = value;
				this.telephone = telephone;
			}
		}

		public virtual string Mobile
		{
			get
			{
				return mobile;
			}
			set
			{
				string mobile = value;
				this.mobile = mobile;
			}
		}

		public virtual string Fax
		{
			get
			{
				return fax;
			}
			set
			{
				string fax = value;
				this.fax = fax;
			}
		}

		public virtual string Email
		{
			get
			{
				return email;
			}
			set
			{
				string email = value;
				this.email = email;
			}
		}

		public virtual string Department
		{
			get
			{
				return department;
			}
			set
			{
				string department = value;
				this.department = department;
			}
		}

		public virtual string Address
		{
			get
			{
				return address;
			}
			set
			{
				string address = value;
				this.address = address;
			}
		}

		public virtual bool Enabled
		{
			get
			{
				return enabled;
			}
			set
			{
				bool enabled = value;
				this.enabled = enabled;
			}
		}

		public virtual string Username
		{
			get
			{
				return username;
			}
			set
			{
				string username = value;
				this.username = username;
			}
		}

		public virtual string Password
		{
			get
			{
				return password;
			}
			set
			{
				string password = value;
				this.password = password;
			}
		}

		public virtual IList<int?> Roles
		{
			get
			{
				return roles;
			}
			set
			{
				IList<int?> roles = value;
				this.roles = roles;
			}
		}
	}
}
